### DISCUSSION TITLE
Error while running merge_batch.csh

### DISCUSSION THREAD

**Discussion by galuhqfs** [gmtsar/gmtsar#365] [Q&A]
Dear, all.

I have a problem when running the `merge_batch.csh` `merge_list` `batch_tops.config`. The error I got is:

cp: cannot stat '..//mnt/f/Skripsi/Data/Cimandiri2/asc/F1/intf_all/2020037_2020061/S1_20200207_ALL_F1.PRM': No such file or directory
grep: ..//mnt/f/Skripsi/Data/Cimandiri2/asc/F1/intf_all/2020037_2020061/S1_20200207_ALL_F1.PRM: No such file or directory

 error -  could not open file supermaster.PRM

grep: supermaster.PRM: No such file or directory
grep: ..//mnt/f/Skripsi/Data/Cimandiri2/asc/F1/intf_all/2020037_2020061/S1_20200207_ALL_F1.PRM: No such file or directory
cp: cannot stat 'supermaster.PRM': No such file or directory
cp: cannot stat '..//mnt/f/Skripsi/Data/Cimandiri2/asc/F2/intf_all/2020037_2020061/S1_20200207_ALL_F2.PRM': No such file or directory
grep: ..//mnt/f/Skripsi/Data/Cimandiri2/asc/F2/intf_all/2020037_2020061/S1_20200207_ALL_F2.PRM: No such file or directory

May I know how to solve it? Thank you

**Comment by Xiaohua-Eric-Xu** [comment]
I think you probably mixed relative and absolute path in your `merge_list`. Maybe it should be ../F1/intf_all/2020037_2020061/S1_20200207_ALL_F1.PRM instead of ..//mnt/f/Skripsi/Data/Cimandiri2/asc/F1/intf_all/2020037_2020061/S1_20200207_ALL_F1.PRM

**Reply by galuhqfs** [reply]
I already rerun the merge_list using the new format `../F1/intf_all/2020037_2020061/:S1_20200207_ALL_F1.PRM:S1_20200302_ALL_F1.PRM,../F2/intf_all/2020037_2020061/:S1_20200207_ALL_F2.PRM:S1_20200302_ALL_F2.PRM`. It worked. But I still have a problem. The result of this process has an unknown error. When I opened my `corr.grd` of a pair, there is some part of area that doesn't contain anything. Is it because the merge_list format?
![corr](https://user-images.githubusercontent.com/106435027/178394944-68095449-0a98-42ca-89ce-b19e8a4ec9c1.png)

**Reply by Xiaohua-Eric-Xu** [reply]
Could you run a `gmt grdinfo phasefilt.grd` in each of the F?/intf_all/2020037_2020061 directory? And also attach the terminal output when you run merge_batch.csh.

**Reply by galuhqfs** [reply]
I already run a `gmt grdinfo phasefilt.grd` and re-run the `merge_batch.csh`. Yet, I still got the problem. The same problem with zero information in some area. Here is the output terminal when I run the merge :
Merging START
Number of Files to be merged is 2
Reading in the grids...
Writing the grid files..Size(6511x13533)...
Stitching location n1 = 117
Number of Files to be merged is 2
Reading in the grids...
Writing the grid files..Size(6511x13533)...
Stitching location n1 = 117
Number of Files to be merged is 2
Reading in the grids...
Writing the grid files..Size(6511x13533)...
Stitching location n1 = 117
Merging END

Recomputing the projection LUT...
hermite: interpolation not in center interval
hermite: interpolation not in center interval
hermite: interpolation not in center interval
hermite: interpolation not in center interval
.
.
.
hermite: interpolation not in center interval
hermite: interpolation not in center interval
hermite: interpolation not in center interval
hermite: interpolation not in center interval
hermite: interpolation not in center interval

SKIP UNWRAP PHASE
rm: No match.

**Reply by Xiaohua-Eric-Xu** [reply]
Feels like something's incorrect with the orbit or the dem.grd files. What if you run `gmt grdinfo dem.grd`?

**Reply by galuhqfs** [reply]
The result of `gmt grdinfo dem.grd` is 

dem.grd: Title: Produced by grdmath
dem.grd: Command: grdmath dem_ortho.grd egm96.grd ADD = dem.grd
dem.grd: Remark:
dem.grd: Gridline node registration used [Geographic grid]
dem.grd: Grid file format: nf = GMT netCDF format (32-bit float), CF-1.7
dem.grd: x_min: 105.7 x_max: 108.3 x_inc: 0.000833333333333 (3 sec) name: longitude n_columns: 3121
dem.grd: y_min: -7.8 y_max: -6.04 y_inc: 0.000833333333333 (3 sec) name: latitude n_rows: 2113
dem.grd: z_min: -7.81462669373 z_max: 3026.49145508 name: z
dem.grd: scale_factor: 1 add_offset: 0
dem.grd: format: netCDF-4 chunk_size: 131,133 shuffle: on deflation_level: 3

Is there something wrong with my dem.grd file?

**Reply by Xiaohua-Eric-Xu** [reply]
Hmmm, it looks OK to me. Are you able to plot the individual phasefilt.grd files in each subswaths and the merge folder? It's hard to guess the problem without earlier information.

**Reply by galuhqfs** [reply]
![phasefilt_F1](https://user-images.githubusercontent.com/106435027/178662571-f43bfae3-6e05-49e1-8531-97622c6b9844.png)
![phasefilt_F2](https://user-images.githubusercontent.com/106435027/178663437-f558eae0-9218-444e-a20f-9ce19697aa3e.png)
![phasefilt_merge](https://user-images.githubusercontent.com/106435027/178663455-da8892ea-9ff0-4ccf-80a1-ea4a7254c3b1.png)

I think something is wrong when I run the interferogram on F2. It's a bit strange that the interferogram result of F2 subswath only on the bottom area. I'll try to re-run it and will inform you later

**Reply by Xiaohua-Eric-Xu** [reply]
I am a bit confused too here. Probably will have to see the data to figure out. A visual inspection is that whether the results can be stitched together? You could run `geocode.csh 0.1` inside each F?/intf_all/day1_day2/ directory, and open the kml files in Google Earth and see whether the products make sense to you, and whether they could be merged together properly.

**Reply by galuhqfs** [reply]
Dear, Xu

I finally knew that my F2 subswath has some errors, so the result of the merge process was error. I runned for creating a new interferogram for F2 and it solved. It covered all of my study area. Thank you for your help

**Reply by Xiaohua-Eric-Xu** [reply]
Great!
